src/DGmethods/DGmodel.jl

```diff
@@ -322,33 +434,25 @@ function (dg::DGModel)(dQdt, Q, ::Nothing, t; increment = false)
         grid.vmap⁻,
         grid.vmap⁺,
         grid.elemtobndy,
-        topology.realelems;
-        ndrange = ndrange_surface,
-        dependencies = (event,),
+        grid.exteriorelems;
+        ndrange = ndrange_exterior_surface,
+        dependencies = (
+            comp_stream,
+            exchange_Q,
```

Comment:
I was surprised by these waits, so adding a comment above would be good.

Reply:
I added comments to the waits.  Hopefully this helps.